### DISCUSSION TITLE
workflow.revision and workflow.commitId return "null"

### DISCUSSION THREAD

**Discussion by rodrigotoro** [nextflow-io/nextflow#3155] [Q&A] [answered]
Hi,

I'm running a workflow locally and would like to use `workflow.revision` and `workflow.commitId` as inside a script in the workflow. The workflow itself lives inside of a git repo, therefore I expect that these properties return the current branch name and commit ID of the repo respectively (as per the Nextflow documentation, https://www.nextflow.io/docs/latest/metadata.html).

However, when I run my workflow both of these properties always return "null". Why is this? Is this the expected behaviour?

Thanks

**Comment by jorgeaguileraseqera** [comment, marked as answer]
These properties are null if the pipeline is executed without a protocol at the beginner. 

I mean, if I have a git repository `myrepo` in my local, it's different to run

`nextflow run ./myrepo`

to

`nextflow run file:///home/jorge/nextflow/myrepo`

without the protocol, the directory is used as a local project but when the protocol is specified it's loaded via git and you can access to these properties
